tools/pythonpkg/src/numpy/array_wrapper.cpp

```diff
@@ -572,26 +572,45 @@ static bool ConvertColumnRegular(NumpyAppendData &append_data) {
 }
 
 template <class DUCKDB_T>
-static bool ConvertDecimalInternal(NumpyAppendData &append_data, double division) {
+static PyObject *ConstructDecimal(DUCKDB_T value, uint8_t width, uint8_t scale, char *buffer) {
+	auto decimal_length = DecimalToString::DecimalLength<DUCKDB_T>(value, width, scale);
+
+	auto &python_import_cache = *DuckDBPyConnection::ImportCache();
+	auto decimal = python_import_cache.decimal.Decimal();
+
+	DecimalToString::FormatDecimal<DUCKDB_T>(value, width, scale, buffer, UnsafeNumericCast<idx_t>(decimal_length));
+	auto stringified = string(buffer, decimal_length);
+	auto py_obj = decimal(py::str(stringified));
```

Comment:
Another way to construct a `decimal.Decimal` is by using the `tuple` constructor:

Example to create a decimal `1.234`:
`decimal.Decimal((0, (1, 2, 3, 4), -3))`

That would require duplicating a lot of the decimal conversion logic to create the digits tuple `(1, 2, 3, 4)` and involves creating even more python objects.
I haven't tested this but I can't imagine this being faster than the current approach